grants_tagger/train_with_sagemaker.py

```diff
@@ -141,3 +148,100 @@ def train_with_sagemaker(instance_type="local", config_version=None, **kwargs):
+        None, help="path to label binarizer"
+    ),
+    model_path: Optional[Path] = typer.Argument(
+        None, help="path to output model.pkl or dir to save model"
+    ),
+    approach: str = typer.Option("tfidf-svm", help="tfidf-svm, scibert, cnn, ..."),
+    parameters: str = typer.Option(
+        None, help="model params in sklearn format e.g. {'svm__kernel: linear'}"
+    ),
+    threshold: float = typer.Option(None, help="threshold to assign a tag"),
+    data_format: str = typer.Option(
+        "list",
+        help="format that will be used when loading the data. One of list,generator",
+    ),
+    train_info: str = typer.Option(None, help="path to train times and instance"),
+    sparse_labels: bool = typer.Option(
+        False, help="flat about whether labels should be sparse when binarized"
+    ),
+    cache_path: Optional[Path] = typer.Option(
+        None, help="path to cache data transformartions"
+    ),
+    config: Path = None,
+    instance_type: str = typer.Option(
+        "local", help="instance type to use when training with Sagemaker"
+    ),
+):
+
+    start = time.time()
+    params_path = os.path.join(os.path.dirname(__file__), "../params.yaml")
+    with open(params_path) as f:
+        params = yaml.safe_load(f)
+
+    # If parameters not provided from user we initialise from DVC
+    if not parameters:
+        parameters = params["train"].get(approach)
+        parameters = convert_dvc_to_sklearn_params(parameters)
+        parameters = str(parameters)
+
+    # Note that config overwrites parameters for backwards compatibility
+    if config:
+        cfg = configparser.ConfigParser(allow_no_value=True)
+        cfg.read(config)
+
+        config_version = cfg["DEFAULT"]["version"]
+        data_path = cfg["data"]["train_data_path"]
+        label_binarizer_path = cfg["model"]["label_binarizer_path"]
+        approach = cfg["model"]["approach"]
+        parameters = cfg["model"]["parameters"]
+        model_path = cfg["model"].get("model_path", None)
+        threshold = cfg["model"].get("threshold", None)
+        if threshold:
+            threshold = float(threshold)
+        data_format = cfg["data"].get("data_format", "list")
+        sparse_labels = cfg["model"].get("sparse_labels", False)
+        if sparse_labels:
+            sparse_labels = bool(sparse_labels)
+        cache_path = cfg["data"].get("cache_path")
+
+    if not config:
+        config_version = None
+    train_with_sagemaker(
+        data_path=data_path,
+        label_binarizer_path=label_binarizer_path,
+        approach=approach,
+        parameters=parameters,
+        model_path=model_path,
+        threshold=threshold,
+        data_format=data_format,
+        sparse_labels=sparse_labels,
+        cache_path=cache_path,
+        instance_type=instance_type,
+        config_version=config_version,
+    )
+
+    duration = time.time() - start
+    print(f"Took {duration:.2f} to train")
```

Comment:
This was supposed to measure how much time train took. It should go back to the main train in cli.py

Reply:
Reason I moved it in both train scripts is so it measures it also when you call the individual scripts, doesn't it make more sense like that?

Reply:
Actually it mostly makes sense for train that happens locally so remove it from sagemaker and only have it in train_cli. Also in train_cli you do not pass the train_info. We want to save this information in a file and you are only printing in screen